### PR TITLE
Add Mce log plugin

### PR DESCRIPTION
**Pull request by MichalRebisz**
Intel Mcelog plugin https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_mcelog

### CODE REVIEW

File: manifests/plugin/mcelog.pp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  Optional[Boolean]         $memory                       = undef,
+  Optional[String]          $mcelogclientsocket           = undef,
+  Optional[Boolean]         $persistentnotification       = undef,
+) {
+
+  include collectd
+
+  collectd::plugin { 'mcelog':
+    ensure  => $ensure,
+    content => template('collectd/plugin/mcelog.conf.erb'),
```
Comment (mrunge) [inline manifests/plugin/mcelog.pp]: where is this file?

File: README.md [hunk 2]
```diff
@@ -132,6 +132,7 @@ documentation for each plugin for configurable attributes.
 * `hddtemp` (see [collectd::plugin::hddtemp](#class-collectdpluginhddtemp) below)
 * `hugepages` (see [collectd::plugin::hugepages](#class-collectdpluginhugepages) below)
 * `intel_pmu` (see [collectd::plugin::intel_pmu](#class-collectdpluginintel_pmu) below)
+* `mcelog` (see [collectd::plugin::mcelog](#class-collectdpluginmcelog) below)
```
Comment (mrunge) [inline README.md]: can you please move this down to keep the list sorted alphabetical?

File: manifests/plugin/mcelog.pp (hunk 1 above)
Comment (MichalRebisz) [reply manifests/plugin/mcelog.pp]: templates/plugin/mcelog.conf.erb

Its the same way intel_pmu is done. Am I incorrect in assuming that the correct path?

File: README.md (hunk 2 above)
Comment (MichalRebisz) [reply README.md]: Moved it down.

File: templates/plugin/mcelog.conf.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<Plugin mcelog>
```
Comment (bastelfreak) [inline templates/plugin/mcelog.conf.erb]: can you please convert this to an epp template? We require this for new templates. For our other review guidelines please have a look at https://voxpupuli.org/docs/reviewing_pr/

File: spec/classes/collectd_plugin_mcelog_spec.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
+require 'spec_helper'
+
+describe 'collectd::plugin::mcelog', type: :class do
+  on_supported_os(baseline_os_hash).each do |os, facts|
+    context "on #{os} " do
+      let :facts do
+        facts
+      end
+
```
Comment (bastelfreak) [inline spec/classes/collectd_plugin_mcelog_spec.rb]: can you please also ensure that this compiles properly?

Something like this should work:

```
it { is_expected.to compile.with_all_deps }
```

Comment (MichalRebisz) [reply spec/classes/collectd_plugin_mcelog_spec.rb]: Added

File: templates/plugin/mcelog.conf.erb (hunk 3 above)
Comment (MichalRebisz) [reply templates/plugin/mcelog.conf.erb]: Converted to epp template

**Pull request merged**